libraries/chain/steem_evaluator.cpp

```diff
@@ -87,7 +87,9 @@ namespace golos { namespace chain {
         void store_account_json_metadata(
             database& db, const account_name_type& account, const string& json_metadata, bool skip_empty = false
         ) {
-#ifndef IS_LOW_MEM
+            if (!db.store_metadata_for_account(account))
+                return;
```

Comment:
Case:
- Set `accounts-to-store-metadata = alice`
- Restart daemon
- Update metadata for alice -> metadata will be saved
- Remove alice from `accounts-to-store-metadata`
- Restart daemon
- Update metadata for alice -> metadata contains old information, 
- There is no way to understand that metadata isn't correct

Resolve:
- Remove old metadata, if it shouldn't be stored